src/escpos/printer/win32raw.py

```diff
@@ -70,36 +77,75 @@ def is_usable() -> bool:
         return is_usable()
 
     @dependency_win32print
-    def __init__(self, printer_name=None, *args, **kwargs):
+    def __init__(self, printer_name: str = "", *args, **kwargs):
         """Initialize default printer."""
         Escpos.__init__(self, *args, **kwargs)
-        if printer_name is not None:
-            self.printer_name = printer_name
-        else:
-            self.printer_name = win32print.GetDefaultPrinter()
-        self.hPrinter = None
-        self.open()
+        self.printer_name = printer_name
+        self.job_name = ""
+
+    @property
+    def printers(self) -> dict:
+        """Available Windows printers."""
+        return {
+            printer["pPrinterName"]: printer
+            for printer in win32print.EnumPrinters(
+                win32print.PRINTER_ENUM_NAME, None, 4
+            )
+        }
 
     @dependency_win32print
-    def open(self, job_name="python-escpos"):
-        """Open connection to default printer."""
-        if self.printer_name is None:
-            raise Exception("Printer not found")
-        self.hPrinter = win32print.OpenPrinter(self.printer_name)
-        self.current_job = win32print.StartDocPrinter(
-            self.hPrinter, 1, (job_name, None, "RAW")
-        )
-        win32print.StartPagePrinter(self.hPrinter)
+    def open(
+        self, job_name: str = "python-escpos", raise_not_found: bool = True
+    ) -> None:
+        """Open connection to default printer.
+
+        By default raise an exception if device is not found.
+
+        :param raise_not_found: Default True.
+                                False to log error but do not raise exception.
+
+        :raises: :py:exc:`~escpos.exceptions.DeviceNotFoundError`
+        """
+        if self._device:
+            self.close()
+
+        self.job_name = job_name
+        try:
+            # Name validation, set default if no given name
+            self.printer_name = self.printer_name or win32print.GetDefaultPrinter()
+            assert self.printer_name in self.printers, "Incorrect printer name"
+            # Open device
+            self.device: Optional[
+                _win32typing.PyPrinterHANDLE
```

Comment:
```suggestion
                "_win32typing.PyPrinterHANDLE"
```
Same here with the import and name.

Reply:
You saved my day ;-D